src/main/java/org/openrewrite/java/testing/mockito/PowerMockitoMockStaticToMockito.java

```diff
@@ -327,154 +297,240 @@ private static List<Expression> getMockedTypesFromPrepareForTestAnnotation(List<
-                          mockedTypesField,
-                          className
-                        ));
-                    }
-                }
-                // The mocked field are needed for other visitors
-                setMockedTypesFields(mockedTypesIdentifiers);
-                return m;
-            }
-            return m;
+            this.setUpMethodAnnotationSignature = "@" + annotationPackage + "." + setUpMethodAnnotationName;
+            this.tearDownMethodAnnotationSignature = "@" + annotationPackage + "." + tearDownMethodAnnotationName;
+
+            this.setUpImportToAdd = annotationPackage + "." + setUpMethodAnnotationName;
+            this.tearDownImportToAdd = annotationPackage + "." + tearDownMethodAnnotationName;
         }
 
-        private void setMockedTypesFields(List<J.Identifier> mockedTypesFields) {
-            this.mockedTypesFields = mockedTypesFields;
+        private Map<J.Identifier, Expression> getMockedTypesFields() {
+            if (mockedTypesFields == null) {
+                mockedTypesFields = getCursor().getNearestMessage(MOCKED_TYPES_FIELDS, new LinkedHashMap<>());
+            }
+            return mockedTypesFields;
         }
 
-        private static boolean isStaticMockAlreadyClosed(J.Identifier staticMock, J.Block methodBody) {
-            return methodBody.getStatements().stream().filter(statement -> statement instanceof J.MethodInvocation)
+        private void findMockStaticInvocations(J.ClassDeclaration classDecl) {
+            // Interfaces have null bodies
+            if (classDecl.getKind() == J.ClassDeclaration.Kind.Type.Interface) {
+                return;
+            }
+            J.Block classBody = classDecl.getBody();
+            List<J.Block> methodBodies = classBody.getStatements().stream()
+              .filter(statement -> statement instanceof J.MethodDeclaration)
+              .map(J.MethodDeclaration.class::cast)
+              .map(J.MethodDeclaration::getBody)
+              .filter(Objects::nonNull)
+              .collect(Collectors.toList());
+            Set<J.MethodInvocation> mockStaticMethodInvocations = new HashSet<>();
+            for (J.Block methodBody : methodBodies) {
+                mockStaticMethodInvocations.addAll(
+                  methodBody.getStatements()
+                    .stream().filter(statement -> statement instanceof J.MethodInvocation)
                     .map(J.MethodInvocation.class::cast)
-                    .filter(MOCKED_STATIC_CLOSE_MATCHER::matches)
-                    .filter(methodInvocation -> methodInvocation.getSelect() instanceof J.Identifier)
-                    .anyMatch(methodInvocation -> ((J.Identifier) methodInvocation.getSelect()).getSimpleName()
-                            .equals(staticMock.getSimpleName()));
+                    .filter(MOCKED_STATIC_MATCHER::matches)
+                    .collect(Collectors.toSet()));
```

Comment:
Iterating over the statements in a method body will miss any statements nested inside of another statement, such as a loop or conditional. 

```
void foo() {
  bar(); // will be found iterating over statements
  if(true) {
    bar(); // will not be found iterating over statements
  }
}
```

Dealing with the nested nature of these hierarchical structures is exactly what the visitor pattern is for. Consider newing up a `JavaVisitor` inline to collect method invocations.